sentry-core/src/hub.rs

```diff
@@ -48,12 +48,14 @@ impl HubImpl {
     }
 
     fn is_active_and_usage_safe(&self) -> bool {
-        let guard = match self.stack.try_read() {
-            Err(TryLockError::Poisoned(err)) => err.into_inner(),
-            Err(TryLockError::WouldBlock) => return false,
+        let guard = match self.stack.read() {
+            Err(err) => err.into_inner(),
             Ok(guard) => guard,
         };
-        guard.top().client.is_some()
+        if let Some(client) = guard.top().client.as_ref() {
```

Comment:
You can shorten this to:

```suggestion
        if let Some(ref client) = guard.top().client {
```

A match:

```suggestion
        match guard.top().client {
            Some(ref client) => client.is_enabled(),
            None => false,
        }
```

Or even a one-liner:

```suggestion
        guard.top().client.as_ref().map_or(false, |c| c.is_enabled())
```

Reply:
ahhh, the expressive power of rust ;-)